packages/openapi-spec/src/openapi-spec-v2.ts

```diff
@@ -100,13 +183,230 @@ export interface InfoObject {
+   */
+  url?: string;
+
+  /**
+   * The email address of the contact person/organization. MUST be in the format
+   * of an email address.
+   */
+  email?: string;
+}
+
+/**
+ * License information for the exposed API.
+ */
+export interface LicenseObject extends Extensible {
+  /**
+   * The license name used for the API.
+   */
+  name: string;
+
+  /**
+   * A URL to the license used for the API. MUST be in the format of a URL.
+   */
+  url?: string;
+}
+
+/**
+ * Allows referencing an external resource for extended documentation.
+ */
+export interface ExternalDocumentationObject extends Extensible {
+  /**
+   * A short description of the target documentation. GFM syntax can be used for
+   * rich text representation.
+   */
+  description?: string;
+
+  /**
+   * The URL for the target documentation. Value MUST be in the format of a URL.
+   */
+  url: string;
+}
+
+/**
+ * Allows adding meta data to a single tag that is used by the Operation Object.
+ * It is not mandatory to have a Tag Object per tag used there.
+ */
+export interface TagObject extends Extensible {
+  /**
+   * The name of the tag.
+   */
+  name: string;
+
+  /**
+   * A short description for the tag. GFM syntax can be used for rich text
+   * representation.
+   */
+  description?: string;
+
+  /**
+   * Additional external documentation for this tag.
+   */
+  externalDocs?: ExternalDocumentationObject;
+}
+
+/**
+ * Allows the definition of a security scheme that can be used by the
+ * operations. Supported schemes are basic authentication, an API key (either
+ * as a header or as a query parameter) and OAuth2's common flows (implicit,
+ * password, application and access code).
+ */
+export interface SecuritySchemeObject extends Extensible {
+  /**
+   * The type of the security scheme. Valid values are "basic", "apiKey" or
+   * "oauth2".
+   */
+  type: 'basic' | 'apiKey' | 'oauth2';
+
+  /**
+   * A short description for security scheme.
+   */
+  description?: string;
+
+  /**
+   * The name of the header or query parameter to be used.
+   */
+  name?: string;
+
+  /**
+   * The location of the API key. Valid values are "query" or "header".
+   */
+  'in'?: 'query' | 'header';
+
+  /**
+   * The flow used by the OAuth2 security scheme. Valid values are "implicit",
+   * "password", "application" or "accessCode".
+   */
+  flow?: 'implicit' | 'password' | 'application' | 'accessCode';
+
+  /**
+   * ("implicit", "accessCode")	Required. The authorization URL to be used for
+   * this flow. This SHOULD be in the form of a URL.
+   */
+  authorizationUrl?: string;
+
+  /**
+   * ("password", "application", "accessCode") Required. The token URL to be
+   * used for this flow. This SHOULD be in the form of a URL.
+   */
+  tokenUrl?: string;
+
+  /**
+   * The available scopes for the OAuth2 security scheme.
+   */
+  scopes?: ScopesObject;
+}
+
+export interface MappingObject<T> {
+  /**
+   * Maps between a name and object
+   */
+  [name: string]: T;
```

Comment:
Are extensions (`x-my-key`) allowed in `MappingObject`? If they are, then I think this value should have type `T | ExtensionValue`.